scripts/prow-smoke-test.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+#================================================================
+# HEADER
+#================================================================
+#%
+#%OVERVIEW 📑
+#+    ./scripts/prow-smoke-test.sh [--validate --preview]
+#%
+#%DESCRIPTION 💡
+#%    Validates openshift-docs AsciiDoc source files using the same tools that run in the Prow CI
+#%
+#%OPTIONS ⚙️
+#%    -v, --validate                                    Validate the AsciiDoc source
+#%    -p, --preview $DISTRO "$PRODUCT_NAME" $VERSION    Use --preview to run with default options
+#%    -h, --help                                        Print this help
+#%
+#%EXAMPLES 🤔
+#%    ./scripts/prow-smoke-test.sh --validate
+#%    ./scripts/prow-smoke-test.sh --preview
+#%    ./scripts/prow-smoke-test.sh --preview openshift-rosa
+#%    ./scripts/prow-smoke-test.sh --preview openshift-pipelines "Red Hat OpenShift Pipelines" 1.14
+#================================================================
+# END_OF_HEADER
+#================================================================
+
+set -e
+
+TEST=$1
+DISTRO=$2
+PRODUCT_NAME=$3
+VERSION=$4
+CONTAINER_IMAGE=quay.io/redhat-docs/openshift-docs-asciidoc
+SCRIPT_HEADSIZE=$(head -30 ${0} |grep -n "^# END_OF_HEADER" | cut -f1 -d:)
+
+if [[ "$1" == "--help" || "$1" == "-h" ]]; then
+    usage
+fi
+
+# Assign default variables
+: ${PRODUCT_NAME:="OpenShift Container Platform"}
+: ${VERSION:="4.16"}
+: ${DISTRO:="openshift-enterprise"}
+
+# Allow podman or docker
+if hash podman 2>/dev/null; then
+    CONTAINER_ENGINE=podman
+elif hash docker 2>/dev/null; then
+    CONTAINER_ENGINE=docker
+else
+    echo >&2 "Container engine not installed. Install Podman or Docker and run the script again."
+    exit 1
+fi
+
+echo "CONTAINER_ENGINE=$CONTAINER_ENGINE 🐳"
+
+# Get the default container $WORKDIR
+CONTAINER_WORKDIR=$($CONTAINER_ENGINE run --rm $CONTAINER_IMAGE /bin/bash -c 'echo $PWD')
+
+# Grep the commented script preamble and return it as help
+display_help() { head -${SCRIPT_HEADSIZE:-99} ${0} | grep -e "^#[%+-]" | sed -e "s/^#[%+-]//g" ; }
+
+# Exit and show the help if no parameters are passed
+if [ $# -eq 0 ]; then
```

Comment:
**nitpick:** We should move this block up top. Otherwise, the script downloads the image first and then shows the help.